rdr_service/api/nph_intake_api.py

```diff
@@ -0,0 +1,216 @@
+                name=activity_name,
+                source=activity_source
+            )
+
+        except KeyError as e:
+            return BadRequest(f'Key error on activity lookup: {e} bundle_id: {self.bundle_identifier}')
+
+    def get_site_id(self, entry: dict):
+        try:
+            pairing_site_code = entry['resource']['type'][0]['coding'][0]['code']
+
+            if not pairing_site_code:
+                raise BadRequest(f'Cannot find site pairing code: bundle_id: {self.bundle_identifier}')
+
+            site = self.nph_site_dao.get_site_id_from_external(external_id=pairing_site_code)
+
+            if not site:
+                raise BadRequest(f'Cannot find site from site code: bundle_id: {self.bundle_identifier}')
+
+            return site.id
+
+        except KeyError as e:
+            raise BadRequest(f'Key error on site lookup: {e} bundle_id: {self.bundle_identifier}')
+
+    def get_event_type_id(self, *, activity_name, activity_source):
+        event_type_dao_instance_items = {k: v for k, v in self.__dict__.items() if 'type_dao' in k}
+
+        if activity_name not in event_type_dao_instance_items.keys():
+            return 1
+
+        event_type_dao = event_type_dao_instance_items[f'nph_{activity_name}_type_dao']
+        event_activity = event_type_dao.get_event_by_source_name(source_name=activity_source)
+
+        if not event_activity:
+            raise BadRequest(f'Cannot find event type: bundle_id: {self.bundle_identifier}')
+
+        return event_activity.id
+
+    def build_event_dao_map(self):
+        event_dao_map = {}
+        event_dao_instance_items = {k: v for k, v in self.__dict__.items() if 'event_dao' in k}
+        for activity in self.current_activities:
+            event_dao_map[f'{activity.name.lower()}'] = event_dao_instance_items[
+                f'nph_{activity.name.lower()}_event_dao']
+        return event_dao_map
+
+    @classmethod
+    def extract_participant_id(cls, participant_obj: dict) -> str:
+        participant_id = participant_obj['resource']['identifier'][0]['value']
+        participant_id = participant_id.split('/')[-1]
+        return participant_id
+
+    def extract_authored_time(self, entry: dict):
+        try:
+            date_time = entry['resource'].get('dateTime')
+
+            if not date_time and entry['resource'].get('period'):
+                date_time = entry['resource']['period']['start']
+
+            if not date_time:
+                raise BadRequest(f'Cannot get value on authored time lookup bundle_id: {self.bundle_identifier}')
+
+            return date_time
+
+        except KeyError as e:
+            raise BadRequest(f'KeyError on authored time lookup: {e} bundle_id: {self.bundle_identifier}')
+
+    @auth_required([RTI, RDR])
+    def post(self):
+        intake_payload = request.get_json(force=True)
+        intake_payload = [intake_payload] if type(intake_payload) is not list else intake_payload
```

Comment:
Please use `intake_payload = [intake_payload] if isintance(intake_payload, list) else intake_payload`

Reply:
Type works fine for this.

Reply:
`isinstance` is more performant in python versions < 3.11. This is fixed in `3.11` AFAIK.

Reply:
Performance shouldn't be an issue in checking if something is a list or not. Thanks